synapsis/src/main/java/com/soen/synapsis/appuser/oauth/CustomOAuth2User.java

```diff
@@ -0,0 +1,43 @@
+package com.soen.synapsis.appuser.oauth;
+
+import com.soen.synapsis.appuser.AppUser;
+import org.springframework.security.core.GrantedAuthority;
+import org.springframework.security.core.authority.SimpleGrantedAuthority;
+import org.springframework.security.oauth2.core.user.OAuth2User;
+
+import java.util.*;
+
+public class CustomOAuth2User implements OAuth2User {
+    private AppUser appUser;
+
+    public CustomOAuth2User(AppUser appUser) {
+        this.appUser = appUser;
+    }
+
+    @Override
+    public Map<String, Object> getAttributes() {
```

Comment:
I'm assuming this is how OAuth treats attribute fetching, but it is inconsistent with how we have implemented it for our users that sign up through the website. We should either find a way to change this to match ours, or change ours to match this. (i.e implement getAttributes in AppUser)

Reply:
We're not using these methods directly. They're used by the framework and must maintain this signature. CustomOAuth2User is effectively a wrapper for AppUser for the OAuth authentication to communicate with.